packages/react/src/components/FormattingToolbar/DefaultButtons/TextAlignButton.tsx

```diff
@@ -1,57 +1,133 @@
 import {
+  Block,
+  BlockFromConfig,
   BlockNoteEditor,
   BlockSchema,
+  defaultProps,
   DefaultProps,
-  PartialBlock,
+  InlineContentSchema,
+  StyleSchema,
 } from "@blocknote/core";
 import { useCallback, useMemo } from "react";
-import type { IconType } from "react-icons";
+import { IconType } from "react-icons";
 import {
   RiAlignCenter,
   RiAlignJustify,
   RiAlignLeft,
   RiAlignRight,
 } from "react-icons/ri";
 
-import { ToolbarButton } from "../../../components-shared/Toolbar/ToolbarButton";
+import { useBlockNoteEditor } from "../../../editor/BlockNoteContext";
 import { useSelectedBlocks } from "../../../hooks/useSelectedBlocks";
+import { ToolbarButton } from "../../../components-shared/Toolbar/ToolbarButton";
 
 type TextAlignment = DefaultProps["textAlignment"];
 
+type BlockConfigWithTextAlignment = {
+  type: string;
+  propSchema: {
+    textAlignment: {
+      default: TextAlignment;
+      values: typeof defaultProps.textAlignment.values;
+    };
+  };
+  content: "inline" | "table" | "none";
+};
+
+const checkTextAlignmentValues = (
+  values: readonly string[]
+): values is typeof defaultProps.textAlignment.values => {
+  return (
+    values.length === defaultProps.textAlignment.values.length &&
+    values.every((value) =>
+      defaultProps.textAlignment.values.includes(value as TextAlignment)
+    )
+  );
+};
+
+function checkBlockTypeHasTextAlignment(
```

Comment:
This check feels a little overkill, i.e. 50+ lines of code to check if textalignment is supported, I'm not sure this is very maintainable. Do you think we can / should simplify this?